### PR TITLE
P2022-47-add-koin-library-dependency->P2022-100-add-koin-dependency-and-call-it

### PR DESCRIPTION
**Pull request by kmaslowiec**

### CODE REVIEW

**Review by Nataniel-Antosik — APPROVED** [review]
In my opinion, everything is good

File: app/src/main/java/com/intive/patronage/retro/MainActivity.kt [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
 
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.util.Log
+import androidx.appcompat.app.AppCompatActivity
+import org.koin.androidx.viewmodel.ext.android.viewModel
 
 class MainActivity : AppCompatActivity() {
+
+    //lazy delegate
```
Comment (MarekMacko) [inline app/src/main/java/com/intive/patronage/retro/MainActivity.kt:10]: Avoid unnecessary comments, it doesn't provide any relevant information to the code. Check other places as well.

Comment (kmaslowiec) [reply app/src/main/java/com/intive/patronage/retro/MainActivity.kt]: Done

**Comment by catchanus** [comment]
The ktlint with the .editorconfig file is already on the develop branch. When merged with a koin branch, ktlint throws errors like "File must end with a newline (\ n)", "Needless blank line", and "Imports must be ordered in lexicographic order without any empty lines ...". It just needs to be corrected.

**Comment by BorysChajdas** [comment]
1. Please look at commits, there are 4. As we agreed there should be always only 1 commit. You need to rebase this branch.
2. Ktlint check is failing, you should correct the code to meet ktlint requirements, look at the logs: https://github.com/intive/patronage22-android-szczecin/runs/4859188571?check_suite_focus=true .

File: app/build.gradle [hunk 2]
```diff
@@ -33,8 +33,11 @@ android {
 
 dependencies {
 
+    // Koin
+    implementation "io.insert-koin:koin-android:3.1.5"
+
     implementation 'androidx.core:core-ktx:1.7.0'
-    implementation 'androidx.appcompat:appcompat:1.4.0'
+    implementation 'androidx.appcompat:appcompat:1.4.1'
```
Comment (BorysChajdas) [inline app/build.gradle:40]: Minor: Such changes are not related to this subtask, it would be better to create another subtask in this task, and to open a new pull request with these changes.

File: app/build.gradle [hunk 3]
```diff
@@ -33,8 +33,11 @@ android {
 
 dependencies {
 
+    // Koin
```
Comment (BorysChajdas) [inline app/build.gradle:36]: Is this really needed :D? Does it provide any useful pieces of information?

File: app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -3,12 +3,14 @@
     package="com.intive.patronage.retro">
 
     <application
-        android:allowBackup="true"
+        android:fullBackupOnly="true"
```
Comment (BorysChajdas) [inline app/src/main/AndroidManifest.xml:6]: Same situation as in my previous comment. It's great that you are fixing some warnings etc. but prefer to open another subtask for such changes. This way it's easier to review by other team members and in case of any issues, it will be easier to revert such changes.

File: app/build.gradle (hunk 2 above)
Comment (kmaslowiec) [reply app/build.gradle]: I have reverted that

File: app/build.gradle (hunk 3 above)
Comment (kmaslowiec) [reply app/build.gradle]: i have removed it

File: app/src/main/AndroidManifest.xml (hunk 4 above)
Comment (kmaslowiec) [reply app/src/main/AndroidManifest.xml:6]: Reverted

File: README.md [hunk 5]
```diff
@@ -1,7 +1,9 @@
...
+
 [SUB-TASK-NUMBER]-[SUB-TASK-NAME]
 `P2022-58-add-README-file`
 
 ## Branch Commit
+
 [MAIN-TASK-NUMBER]-[MAIN-TASK-NAME] -> [SUB-TASK-NUMBER]-[SUB-TASK-NAME]
-`P2022-57 Add Readme -> P2022-58 add README file`
\ No newline at end of file
+`P2022-57 Add Readme -> P2022-58 add README file`
```
Comment (MarekMacko) [inline README.md:9]: Above changes shouldn't be placed in this PR

**Comment by MarekMacko** [comment]
Pull request title seems to be outdated, it should be the same as the commit message
